app/models/user.rb

```diff
@@ -14,7 +14,8 @@ class User < ApplicationRecord
   mount_uploader :avatar, AvatarUploader
 
   belongs_to :osbb, optional: true
-
+  validates_associated :osbb
+  accepts_nested_attributes_for :osbb
   has_one :address, as: :addressable, dependent: :destroy
```

Comment:
better to store associations  and validations in one place, you have belongs_to, then validates_associated, then accepts_nested_attributes_for, then associations, and after validations

Reply:
later it will be hard to manage that is where